validation-test/stdlib/Dictionary.swift

```diff
@@ -99,10 +99,21 @@ DictionaryTestSuite.test("Hashable") {
     let dd7: Dictionary<Int, Dictionary<Int, String>> = [:]
     checkHashable([dd1, dd2, dd3, dd4, dd5, dd6, dd7], equalityOracle: { $0 == $1 })
 
-    // d5 is equal to d4, but will probably be iterated differently
-    let d5: Dictionary<Int, String> = [1: "meow", 4: "mooo", 2: "meow"]
-    expectEqual(d4, d5)
-    expectEqual(d4.hashValue, d5.hashValue)
+    // d5 is equal to d6, but should be iterated differently
+    var d5: Dictionary<Int, String> = Dictionary()
+    d5.reserveCapacity(3)
+    d5[1] = "meow"
+    d5[4] = "mooo"
+    d5[2] = "meow"
+
+    var d6: Dictionary<Int, String> = Dictionary()
+    d6.reserveCapacity(3)
```

Comment:
You should reserve a different amount here; for example, `d6.reserveCapacity(25)` would arrange elements in a different permutation (at least with the hash function we currently have). Dictionaries with the same capacity and elements will arrange elements in the same order.

An even better solution would be to reuse the same dictionary but reserve progressively larger capacities, checking that the hash value does not change after each reservation:

```swift
var d5: Dictionary<Int, String> = [1: "meow", 4: "mooo", 2: "meow"]
let expected = d5.hashValue
for capacity in [4, 8, 16, 32, 64, 128, 256] {
  d5.reserveCapacity(capacity)
  expectEqual(d5.hashValue, expected)
}
```

This makes it a lot more likely that the keys will fall into a different order in at least one iteration. To encourage reordering even more, start with a larger dictionary -- 3 elements have just 6 permutations, while, say, 6 elements have 720.

Reply:
Thanks @lorentey ,
With your suggestion the test now correctly fails on a hashValue implementation that depends on the traversal order.